uber/site_sections/preregistration.py

```diff
@@ -312,22 +316,35 @@ def prereg_payment(self, session, payment_id, stripeToken):
         if message:
             raise HTTPRedirect('index?message={}', message)
 
+        # from this point on, the credit card has actually been charged but we haven't marked anything as charged yet.
+        # be ultra-careful until the attendees/groups are marked paid and written to the DB or we could end up in a
+        # situation where we took the payment, but didn't mark the cards charged
+
         for attendee in charge.attendees:
             attendee.paid = c.HAS_PAID
             attendee.amount_paid = attendee.total_cost
+            log.info("PAYMENT: marked attendee id={} ({},{}) as paid", attendee.id, attendee.first_name, attendee.last_name)
             session.add(attendee)
 
         for group in charge.groups:
             group.amount_paid = group.default_cost
+            log.info("PAYMENT: marked group id={} ({}) as paid", group.id, group.name)
+
             for attendee in group.attendees:
                 attendee.amount_paid = attendee.total_cost - attendee.badge_cost
+                log.info("PAYMENT: marked group member id={} ({},{}) as paid", attendee.id, attendee.first_name,
+                         attendee.last_name)
             session.add(group)
 
         Charge.unpaid_preregs.clear()
         Charge.paid_preregs.extend(charge.targets)
+
+        session.flush()  # paranoia: really make sure we lock in marking taking payments in the database
```

Comment:
we don't need to call session.flush() to save the data but, i figured it might be a good idea to hit save as early as humanly possible in this area of the code where we're marking attendees as paid after taking their payment.

if anyone thinks there could unforseen side effects here, we can remove it.